spec/models/item_spec.rb

```diff
@@ -24,13 +25,76 @@
+      let!(:invoice_6) { create(:invoice, customer_id: customer_1.id, status: 1)}
+      let!(:invoice_item_5) { create(:invoice_item, invoice: invoice_5, item: item_3, unit_price: 600000, quantity: 1) }
+      let!(:invoice_item_6) { create(:invoice_item, invoice: invoice_6, item: item_3, unit_price: 600000, quantity: 1) }
+
+      let!(:item_4) { create(:item, merchant_id: merchant_1.id, status: 1)}
+      let!(:invoice_7) { create(:invoice, customer_id: customer_1.id, status: 1)}
+      let!(:invoice_8) { create(:invoice, customer_id: customer_1.id, status: 1)}
+      let!(:invoice_item_7) { create(:invoice_item, invoice: invoice_7, item: item_4, unit_price: 12, quantity: 1) }
+      let!(:invoice_item_8) { create(:invoice_item, invoice: invoice_8, item: item_4, unit_price: 12, quantity: 1) }
+
+      let!(:item_5) { create(:item, merchant_id: merchant_1.id, status: 1)}
+      let!(:invoice_9) { create(:invoice, customer_id: customer_1.id, status: 1)}
+      let!(:invoice_10) { create(:invoice, customer_id: customer_1.id, status: 1)}
+      let!(:invoice_item_9) { create(:invoice_item, invoice: invoice_9, item: item_5, unit_price: 10000, quantity: 1) }
+      let!(:invoice_item_10) { create(:invoice_item, invoice: invoice_10, item: item_5, unit_price: 10000, quantity: 1) }
+
+      let!(:item_6) { create(:item, merchant_id: merchant_1.id, status: 1)}
+      let!(:invoice_11) { create(:invoice, customer_id: customer_1.id, status: 1)}
+      let!(:invoice_12) { create(:invoice, customer_id: customer_1.id, status: 1)}
+      let!(:invoice_item_11) { create(:invoice_item, invoice: invoice_11, item: item_6, unit_price: 1200000, quantity: 1) }
+      let!(:invoice_item_12) { create(:invoice_item, invoice: invoice_12, item: item_6, unit_price: 1200000, quantity: 1) }
+
+      let!(:transaction_1) { create(:transaction, invoice: invoice_1, result: 0) }
+      let!(:transaction_2) { create(:transaction, invoice: invoice_2, result: 0) }
+      let!(:transaction_3) { create(:transaction, invoice: invoice_3, result: 0) }
+      let!(:transaction_4) { create(:transaction, invoice: invoice_4, result: 0) }
+      let!(:transaction_5) { create(:transaction, invoice: invoice_5, result: 0) }
+      let!(:transaction_6) { create(:transaction, invoice: invoice_6, result: 0) }
+      let!(:transaction_7) { create(:transaction, invoice: invoice_7, result: 0) }
+      let!(:transaction_8) { create(:transaction, invoice: invoice_8, result: 0) }
+      let!(:transaction_9) { create(:transaction, invoice: invoice_9, result: 0) }
+      let!(:transaction_10) { create(:transaction, invoice: invoice_10, result: 0) }
+      let!(:transaction_11) { create(:transaction, invoice: invoice_11, result: 0) }
+      let!(:transaction_12) { create(:transaction, invoice: invoice_12, result: 0) }
+
+      it "displays the top five items by total revenue" do
+        expected = [item_6, item_1, item_3, item_2, item_5]
+        not_expected = [item_4]
```

Comment:
Are you using this anywhere?

Reply:
You're going to have to specify what "this" is here please. I can take out line 92 if you'd like.